src/main/java/frc/robot/Robot.java

```diff
@@ -531,6 +570,78 @@ public void turnControlPanel() {
       m_controlPanelSpinAmount -= 1;
   }
 
+  private void processImage() {
+    m_sink.grabFrame(m_sourceMat);
+    if(doVisionProcessing) {
+      if(!m_sourceMat.empty()) {
+        // Crop out the uneccessary parts of the image
+        Mat m_subMat = new Mat(m_sourceMat, scanArea);
+
+        // Converts m_sourceMat to hsv color
+        Imgproc.cvtColor(m_subMat, m_hsvMat, Imgproc.COLOR_BGR2HSV);
+
+        // Filters for only the color of the tape
+        Core.inRange(m_hsvMat, hsvLow, hsvHigh, m_filteredMat);
+
+        // Find contours
+        Imgproc.findContours(m_filteredMat, contours, new Mat(),
+          Imgproc.RETR_TREE, Imgproc.CHAIN_APPROX_SIMPLE);
+
+        if(contours.size() > 0) {
+          MatOfPoint largestContour = contours.get(0);
+          double largestContourSize = Imgproc.contourArea(largestContour);
+          for(MatOfPoint contour : contours) {
+            double size = Imgproc.contourArea(contour);
+            if(size > largestContourSize) {
+              largestContour = contour;
+              largestContourSize = size;
+            }
+          }
+          contours.clear();
+          contours.add(largestContour);
+
+          Rect bounds = Imgproc.boundingRect(largestContour);
+          Point center = new Point(bounds.x + (bounds.width/2), bounds.y);
+          if(points.size() >= 4) {
+            points.remove(0);
+          }
+          points.add(center);
+          
+          center = new Point(0, 0);
+          for(Point point : points) {
+            center.x += point.x;
+            center.y += point.y;
+          }
+
+          center.x /= points.size();
+          center.y /= points.size();
+          center.y += 120;
+
+          Imgproc.circle(m_sourceMat, center, 5, new Scalar(0, 0, 255), 3);
+          Imgproc.circle(m_sourceMat, new Point(320, center.y), 3, new Scalar(255, 0, 0), -1);
+          Imgproc.rectangle(m_sourceMat, new Point(scanArea.x, scanArea.y), new Point(scanArea.x + scanArea.width, scanArea.y + scanArea.height), new Scalar(0, 255, 0));
+          Imgproc.drawContours(m_sourceMat, contours, -1, new Scalar(255, 0, 0));
```

Comment:
Is there a reason for not just drawing `largestContour`?

Reply:
AFAIK there is no method to draw a single MatOfPoint object.